### PR TITLE
feat(@clayui/dual-list-box): Make reorder & transfer button state more consistent

### LINKED ISSUE #3682: Reorder buttons ClayDualListBox
-   [X] I have searched the [issues](https://github.com/liferay/clay/issues) of this repository and believe that this is not a duplicate.

## Is there an example you can provide via codesandbox.com?

* https://clayui.com/docs/components/dual-list_box.html
* https://codesandbox.io/s/clayduallistbox-content-dashboard-configuration-0ze8o

Both are examples of `ClayDualListBox` with the out-of-the-box implementation.

## What are the steps to reproduce?

You can view these results with the out-of-the-box component.

## What is the expected result?

1. When there is only one element in the "In Use" box, disable both reorder buttons.
2. When there are two elements, and the user selects the first one, disable the reorder up button.
2. When there are two elements, and the user selects the second one, disable the reorder down button.

### Actual behavior screenshots
1.
Screenshot 2020-09-03 at 17 18 59

2.
Screenshot 2020-09-03 at 17 19 10

3.
Screenshot 2020-09-03 at 17 19 14

I don't know if it's a bug or an improvement. But I think it would be very nice to have it.

Thanks! 😄

### PR DESCRIPTION
**Pull request by kresimir-coko**
Fixes #3682 

I believe I've handled all cases of buttons being disabled except for one, and that is something has been transferred between select boxes but nothing is visually selected. After investigating, I noticed that the transferred item (or items if more have been selected), is kept in the `leftSelected` variable (in the case when transfering from left to right). I couldn't figure out a clean way of resetting the `leftSelected` value back to an empty array.

### CODE REVIEW

File: packages/clay-form/stories/index.tsx [hunk 1]
```diff
@@ -87,11 +87,15 @@ storiesOf('Components|ClayDualListBox', module).add('default', () => {
 		<ClayDualListBox
 			disableLTR={
 				boolean('disableLTR', false) ||
-				secondSelectBoxItems.length >= rightMaxItems
+				secondSelectBoxItems.length >= rightMaxItems ||
+				!firstSelectBoxItems.length ||
+				!leftSelected.length
```
Comment (bryceosterhaus) [inline packages/clay-form/stories/index.tsx:92]: I don't think we shouldn't put this responsibility on the user to manage these checks, I think we should be able to internalize.

File: packages/clay-form/src/SelectBox.tsx [hunk 2]
```diff
@@ -212,7 +214,13 @@ const ClaySelectBox: React.FunctionComponent<IProps> = ({
...
 								aria-label={ariaLabels.reorderUp}
 								className="reorder-button reorder-button-up"
-								disabled={!value.length}
+								disabled={
+									!value.length ||
+									selectedIndex === 0 ||
+									selectedIndexes.includes(0) ||
+									!items.length ||
+									selectedIndexes.length === 0
+								}
```
Comment (bryceosterhaus) [inline packages/clay-form/src/SelectBox.tsx:223]: This expression is hard to follow, I think it'd be best to put into some named variables in order to clean it up. I also noticed that `selectedIndexes` is directly related to `value`, so you shouldn't need to have checks for both.

```jsx
const noItems = !items.length;
const noItemsSelected = !selectedIndexes.length;
const firstItemSelected = selectedIndex === 0 || selectedIndexes.includes(0);
```

**Review by bryceosterhaus — CHANGES_REQUESTED** [review]

File: packages/clay-form/stories/index.tsx (hunk 1 above)
Comment (kresimir-coko) [reply packages/clay-form/stories/index.tsx:92]: Yea, that makes sense, I'll get on top of that!

File: packages/clay-form/src/SelectBox.tsx (hunk 2 above)
Comment (kresimir-coko) [reply packages/clay-form/src/SelectBox.tsx:223]: You're right, I got caught up in making it work, so I didn't think much about readability.

**Comment by kresimir-coko** [comment]
@bryceosterhaus I've sent a commit that should cover your requests

File: packages/clay-form/src/SelectBox.tsx [hunk 3]
```diff
@@ -154,6 +154,19 @@ const ClaySelectBox: React.FunctionComponent<IProps> = ({
...
+	const noItems = !items.length;
+
+	const noItemsSelected = !selectedIndexes.length;
+
+	const firstItemSelected =
+		selectedIndex === 0 || selectedIndexes.includes(0);
+
+	const lastItemSelected =
+		selectedIndex === items.length - 1 ||
+		selectedIndexes.includes(items.length - 1);
```
Comment (bryceosterhaus) [inline packages/clay-form/src/SelectBox.tsx:168]: I don't think you need the `selectedIndex` variable. Can't this just be simplified to this?

```suggestion
	const noItems = !items.length;
	const noItemsSelected = !selectedIndexes.length;
	const firstItemSelected = selectedIndexes.includes(0);
	const lastItemSelected = selectedIndexes.includes(items.length - 1);
```

Comment (kresimir-coko) [reply packages/clay-form/src/SelectBox.tsx]: It can! @bryceosterhaus good job spotting that, because `selectedIndexes` is always an array, even with just one selected item, `selectedIndexes.includes()` applies always.

**Comment by kresimir-coko** [comment]
I've pushed even more suggested changes, simplifying the checks and removing the `selectedIndex` variable.

**Review by bryceosterhaus — APPROVED** [review]
LGTM!
